### PR TITLE
Add travel app

### PR DESCRIPTION
**Pull request by reinbach**
[ch15727]

Notifications will need to be reviewed, but this should be a start for the travel app

### CODE REVIEW

File: src/etools/applications/travel/models.py [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+def generate_reference_number():
+    with transaction.atomic():
+        numeric_part = connection.tenant.counters.get_next_value(
+            WorkspaceCounter.TRAVEL_REFERENCE
+        )
+    year = timezone.now().year
+    return '{}/{}'.format(year, numeric_part)
+
+
+class Itinerary(TimeStampedModel):
```
[the remark this answers is not in the document]
Comment (reinbach) [reply src/etools/applications/travel/models.py]: Nope, I thought `Itinerary` made sense here. Am I missing something?

File: src/etools/applications/travel/models.py [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+    itinerary_illegal_transition,
+    itinerary_rejected,
+    itinerary_submitted,
+    itinerary_subreview,
+)
+from etools.applications.users.models import WorkspaceCounter
+
+
+def generate_reference_number():
+    with transaction.atomic():
```
[the remark this answers is not in the document]
Comment (reinbach) [reply src/etools/applications/travel/models.py]: Are we wanting to change the reference number format?
e.g. `<COUNTRY_SHORT_CODE>/<YEAR>TRAVEL<PK>`

File: src/etools/applications/travel/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (robertavram) [reply src/etools/applications/travel/models.py]: Nik votes to call it a "Trip"

[the remark this answers is not in the document]
Comment (robertavram) [reply src/etools/applications/travel/models.py]: Evan votes for having the model be the same as the UI (pending UI)

[the remark this answers is not in the document]
Comment (robertavram) [reply src/etools/applications/travel/models.py]: I abstain courteously :)

[the remark this answers is not in the document]
Comment (reinbach) [reply src/etools/applications/travel/models.py]: You're a wuss :)

**Pull request closed without merging**